src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compiler.cs

```diff
@@ -87,7 +87,14 @@ internal TextWriter Source
 
         internal static string GetTempAssemblyName(AssemblyName parent, string? ns)
         {
-            return parent.Name + ".XmlSerializers" + (ns == null || ns.Length == 0 ? "" : "." + ns.GetHashCode());
+            return parent.Name + ".XmlSerializers" + (ns == null || ns.Length == 0 ? "" : "." + GetPersistentHashCode(ns));
+        }
+
+        private static uint GetPersistentHashCode(string value)
+        {
+            byte[] valueBytes = Encoding.UTF8.GetBytes(value);
+            byte[] hash = System.Security.Cryptography.SHA512.Create().ComputeHash(valueBytes);
```

Comment:
This is Disposable. Although it doesn't seem to have anything to dispose any more, it will go on the finalize queue otherwise.

Reply:
Thanks,
Looking at the current implementation (HashAlgorithm.cs), I agree with your assessment that ultimately no harm is done here, Stephen also noted that "this is not a hot code path".
This seems to me more readable, but I can add a "using" clause if you feel that's more appropriate / future-proof.